### PR TITLE
lilypond: 2.22.2 -> 2.23.11

### PR DESCRIPTION
**Pull request by usertam**
###### Description of changes

Changes [here](https://lilypond.org/doc/v2.23/Documentation/changes) since 2.22.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yurrriq** [comment]
Thanks! I should be able to test in the next several days.

**Comment by usertam** [comment]
Noticed `2.23.11` just dropped a few days ago, so I added another commit here. Hope that's okay.

**Comment by stigtsp** [comment]
> Noticed `2.23.11` just dropped a few days ago [here](https://lilypond.org/download/sources/v2.23/), so I added another commit here. Hope that's okay.

Hi! Could you squash the two, so we have one commit for lilypond: 2.22.2 -> 2.23.11?

**Comment by usertam** [comment]
@stigtsp It's done!

**Comment by stigtsp** [comment]
Result of `nixpkgs-review pr 182614` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lilypond-unstable</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>solfege</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>clevis</li>
    <li>denemo</li>
    <li>disorderfs</li>
    <li>frescobaldi</li>
    <li>kakounePlugins.rep</li>
    <li>lilypond</li>
    <li>lilypond-with-fonts</li>
    <li>luksmeta</li>
    <li>tang</li>
  </ul>
</details>

@GrahamcOfBorg build solfege
@GrahamcOfBorg build spring springLobby

**Comment by yurrriq** [comment]
We'll want to keep `lilypond` on 1.22, but we can do 1.23 for unstable. Looks like the build is currently failing?

**Comment by usertam** [comment]
> We'll want to keep `lilypond` on 1.22, but we can do 1.23 for unstable. Looks like the build is currently failing?

I think `lilypond-unstable` is currently [dropped][1] right now, but I am happy to bring it back!
But regarding the failed packages, I don't think they are depended on lilypond at all? Please correct me if I am wrong about all this though.

spring:
```sh
$ nix why-depends nixpkgs#spring nixpkgs#lilypond
'/nix/store/gkq7vfs0720543qcx1kvh03zpgcm2nbd-spring-105.0.1-1486-g8581792' does not depend on '/nix/store/ikfg4d7q1jqlw290f2pddjfw1pwybz3l-lilypond-2.22.2'
```

springLobby:
```sh
$ nix why-depends nixpkgs#springLobby nixpkgs#lilypond
'/nix/store/5hy48hpvvamw16iivc59cl6aa15v87a8-springlobby-0.270' does not depend on '/nix/store/ikfg4d7q1jqlw290f2pddjfw1pwybz3l-lilypond-2.22.2'
```

and solfege which failed to build because of missing python headers:
```
$ nix why-depends nixpkgs#solfege nixpkgs#lilypond
error: builder for '/nix/store/a5pr2lpcv4mniw1zhd0cx1xvpw8yl8ar-solfege-3.23.4.drv' failed with exit code 1;
       last 10 log lines:
       > checking fcntl.h presence... yes
       > checking for fcntl.h... yes
       > checking sys/ioctl.h usability... yes
       > checking sys/ioctl.h presence... yes
       > checking for sys/ioctl.h... yes
       > checking for unistd.h... (cached) yes
       > configure: error:
       > *** We need the python header files to build the OSS sound module.
       > *** The debian package is called python3-dev, other distributions
       > *** probably have a similar name.
       For full logs, run 'nix log /nix/store/a5pr2lpcv4mniw1zhd0cx1xvpw8yl8ar-solfege-3.23.4.drv'.
```

[1]: https://github.com/NixOS/nixpkgs/commit/389c7cf0a82de3a6de9d1832a4d99fa66d16d91d

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -34826,7 +34826,7 @@ with pkgs;
 
   lguf-brightness = callPackage ../misc/lguf-brightness { };
 
-  lilypond = callPackage ../misc/lilypond { guile = guile_1_8; };
+  lilypond = callPackage ../misc/lilypond { guile = guile_2_2; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:34829]:
```suggestion
  lilypond = callPackage ../misc/lilypond { };
```
that is the default guile version

**Comment by usertam** [comment]
New PR https://github.com/NixOS/nixpkgs/pull/184327 opened for unstable instead, closing this one.

**Pull request closed without merging**